typescript/src/ethereum.ts

```diff
@@ -446,7 +561,90 @@ class WalletRegistry {
    *          hex string.
    */
   async getWalletPublicKey(walletID: string): Promise<string> {
-    const publicKey = await this._walletRegistry.getWalletPublicKey(walletID)
+    const publicKey = await this._instance.getWalletPublicKey(walletID)
     return publicKey.substring(2)
   }
 }
+
+/**
+ * Implementation of the Ethereum TBTCVault handle.
+ */
+export class TBTCVault extends EthereumContract implements ChainTBTCVault {
+  constructor(config: ContractConfig) {
+    super(config, TBTCVaultDeployment)
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainTBTCVault#requestOptimisticMint}
+   */
+  async requestOptimisticMint(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): Promise<void> {
+    await this._instance.requestOptimisticMint(
+      depositTxHash,
+      depositOutputIndex
+    )
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainTBTCVault#optimisticMintingRequests}
+   */
+  async optimisticMintingRequests(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): Promise<OptimisticMintingRequest> {
+    const depositKey = Bridge.buildDepositKey(depositTxHash, depositOutputIndex)
+
+    const request = await this._instance.optimisticMintingRequests(depositKey)
+
+    return this.parseOptimisticMintingRequest(request)
+  }
+
+  /**
+   * Parses a optimistic minting request using data fetched from the on-chain contract.
+   * @param request Data of the optimistic minting request.
+   * @returns Parsed optimistic minting request.
+   */
+  private parseOptimisticMintingRequest(
+    request: any
+  ): OptimisticMintingRequest {
+    return {
+      requestedAt: BigNumber.from(request.requestedAt).toNumber(),
+      finalizedAt: BigNumber.from(request.finalizedAt).toNumber(),
+    }
+  }
+
+  // TODO: Add docs
+  // TODO: Consider exposing in optimistic-minting.ts and in chain.ts
+  async queryOptimisticMintingRequestedEvents(
+    fromBlock?: string,
```

Comment:
It's a little weird to me that we introduced a `Hex` type and we're still stringly typing stuff all over the place